igvm/hypervisor.py

```diff
@@ -1194,9 +1197,32 @@ def log_migration(self, vm: VM, operator: str) -> None:
         :param operator: plus for migration to HV, minus for migration from HV
         """
 
-        cpu_usage_vm = self.estimate_vm_cpu_usage(vm)
-        timestamp = int(time())
-        log_entry = '{} {}{}'.format(timestamp, operator, round(cpu_usage_vm))
-
-        self.dataset_obj['igvm_migration_log'].add(log_entry)
-        self.dataset_obj.commit()
+        retry = 5
+        for i in range(0, retry):
+            # It can happen, that multiple migrations at the same time will write the same log
+            # entry. This will fail during the commit and throw a DatasetError
+            try:
+                cpu_usage_vm = self.estimate_vm_cpu_usage(vm)
+                timestamp = int(time())
+                log_entry = '{} {}{}'.format(timestamp, operator, round(cpu_usage_vm))
+                self.dataset_obj['igvm_migration_log'].add(log_entry)
```

Comment:
After this the changes are staged for commit so the first retry will add two estimated, the initial and the one from the retry, the second three and so on. I think you don't want this no ?

Reply:
I updated the code. In case the entry already exists, I'm removing it from the set again. The next iteration of the loop, will create a different entry, since the timestamp changed, which then will be unique again.